### PR TITLE
Default silent to platform convention

### LINKED ISSUE #193: Defaulting silent to false could conflict with OS defaults
The `NotificationOptions` dictionary specifies `boolean silent=false;` by default. 

That default can conflict with certain platforms/OSs, whereby if a notification is silent (or not) should really be left up to OS. On the WebKit side, we are wondering if it's feasible to change the default value and semantics of `silent` to be equivalent to "OS decides"?

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #193 

- [x] At least two implementers are interested (and none opposed):
   * WebKit
   * Chromium per https://github.com/whatwg/notifications/pull/194#issuecomment-1551191216
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/WebKit/WebKit/pull/13912
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1446893
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1834005
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=256828
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: https://github.com/mdn/content/issues/26812

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/notifications/194.html" title="Last updated on May 19, 2023, 12:57 PM UTC (a6b4f0c)">Preview</a> | <a href="https://whatpr.org/notifications/194/8b4002d...a6b4f0c.html" title="Last updated on May 19, 2023, 12:57 PM UTC (a6b4f0c)">Diff</a>

### CODE REVIEW

File: notifications.bs [hunk 1]
```diff
@@ -84,7 +84,8 @@ an existing notification.
 
 <p>A <a>notification</a> has an associated
 <dfn for=notification id=silent-preference-flag>silent preference</dfn> boolean, which is initially
```
Comment (annevk) [inline notifications.bs:86]: null or a boolean (might need some general rewording to make that more palatable)

File: notifications.bs [hunk 2]
```diff
@@ -83,9 +83,9 @@ initially false. When true, indicates that the end user should be alerted after
 an existing notification.
 
 <p>A <a>notification</a> has an associated
-<dfn for=notification id=silent-preference-flag>silent preference</dfn> boolean, which is initially
-null. When true, indicates that no sounds or vibrations should be made. When null, indicates that
-producing sounds or vibrations should be left to platform conventions.
+<dfn for=notification id=silent-preference-flag>silent preference</dfn> nullable boolean, which is
```
Comment (annevk) [inline notifications.bs:86]: While interesting, I'd prefer we stick with "null or boolean" or "boolean or null" for the type. We're not quite happy with IDL's ?-convention so I don't think we should start using it in prose.

Comment (marcoscaceres) [reply notifications.bs]: Makes sense.

**Review by annevk — APPROVED** [review]

**Comment by rayankans** [comment]
Thanks for the proposal Marcos, agreed that the default should be platform specific, and is an overall improvement. The change looks good to me, and happy to apply the changes from the chromium side of things.

**Comment by marcoscaceres** [comment]
Implementer bugs filed.

**Comment by marcoscaceres** [comment]
Tests sent at https://github.com/web-platform-tests/wpt/pull/40160 (reviewed upstream)

**Pull request merged**